Jumbo Docs & Cleanup (#35)
htmlcsstoimage/docs: guides/advanced/jumbo-images.md

---
layout: page
title: Jumbo Images
permalink: /guides/advanced/jumbo-images/
parent: Advanced
grand_parent: Guides
nav_order: 6
description: >-
  Render large images without reducing quality by splitting into tiles and stitching them together.
---

# Jumbo Images
{: .no_toc }
{: .fs-9 }

Render large screenshots without reducing quality by splitting and stitching tiles.
{: .fs-6 .fw-300 }

<hr>

1. TOC
{:toc}

<hr>

## What are jumbo images?

Normal images are limited to 8000 x 8000 pixels. If your image scales beyond this, we automatically scale it down by default.

> Jumbo images allow you to scale beyond this. When you set `jumbo_max_width` and `jumbo_max_height`, the API renders lets you generate images up to **80,000 pixels** on a side.

Reviewer: ```suggestion
Jumbo images allow you to scale beyond this. When you set `jumbo_max_width` and `jumbo_max_height`, the API lets you generate images up to **80,000 pixels** on a side.
```

> Rendering really tall or wide html with Puppeteer has issues around <a href="https://github.com/puppeteer/puppeteer/issues/1576" target="_blank">8000px</a> where your content may start repeating or overlapping. Most images don't reach these extremes, and when they do - we'll scale down your image to fit and prevent this behavior.

Reviewer: ```suggestion
Normal images are limited to 8000 x 8000 pixels. If your image scales beyond this, we automatically scale it down by default.
```

> Jumbo images get around this. When you set `jumbo_max_width` and `jumbo_max_height`, the API renders your output in **8000 x 8000 tiles** and stitches them back into a single image. This lets you generate images up to **80,000 pixels** on a side.

Reviewer: ```suggestion
Jumbo images allow you to scale beyond this. When you set `jumbo_max_width` and `jumbo_max_height`, the API renders lets you generate images up to **80,000 pixels** on a side.
```
